src/test/java/lambda/part2/exercise/Exercise2.java

```diff
@@ -25,75 +25,91 @@ public void personHasNotEmptyLastNameAndFirstName() {
     // TODO - возвращает новый предикат, являющийся отрицанием исходного
     // TODO - при реализации использовать логический оператор !
     private Predicate<Person> negateUsingLogicalOperator(Predicate<Person> predicate) {
-        throw new UnsupportedOperationException();
+        return p -> !predicate.test(p);
     }
 
     // TODO метод (Person -> boolean, Person -> boolean) -> (Person -> boolean)
     // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
     // TODO - при реализации использовать логический оператор &&
     private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
-        throw new UnsupportedOperationException();
+        return p -> left.test(p) && right.test(p);
     }
 
     @Test
     public void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
-        Predicate<Person> personHasEmptyFirstName = null;
-        Predicate<Person> personHasEmptyLastName = null;
+        Predicate<Person> personHasEmptyFirstName = negateUsingLogicalOperator(p -> !p.getFirstName().isEmpty());
+        Predicate<Person> personHasEmptyLastName = negateUsingLogicalOperator(p -> !p.getLastName().isEmpty());
 
-        Predicate<Person> personHasNotEmptyFirstName = null;
-        Predicate<Person> personHasNotEmptyLastName = null;
+        Predicate<Person> personHasNotEmptyFirstName = negateUsingLogicalOperator(p -> p.getFirstName().isEmpty());
+        Predicate<Person> personHasNotEmptyLastName = negateUsingLogicalOperator(p -> p.getLastName().isEmpty());
 
-        Predicate<Person> personHasNotEmptyLastNameAndFirstName = null;
+        Predicate<Person> personHasNotEmptyLastNameAndFirstName = p -> !p.getLastName().isEmpty() && !p.getFirstName().isEmpty();
```

Comment:
u have written special function (andUsingLogicalOperator) for this condition, so why not use it ?)

Reply:
because to this line I've already forgotten about that function)